pkg/pool-manager/virtualmachine_pool.go

```diff
@@ -445,51 +540,123 @@ func (p *PoolManager) vmWaitingCleanupLook() {
+			macMapChanged = true
+			logger.Info("entry is stale", "macAddress", macAddress, "vmFullName", macEntry.instanceName, "interfaceName", macEntry.macInstanceKey, "stale TS", macEntry.transactionTimestamp)
+			vm, err := p.getvmInstance(macEntry.instanceName)
+			if err != nil && apierrors.IsNotFound(err) {
+				logger.Info("vm no longer exists or does not have Template. removing mac from pool", "macAddress", macAddress, "entry", macEntry)
+				p.macPoolMap.removeMacEntry(macAddress)
+			} else if err == nil {
+				p.macPoolMap.alignMacEntryAccordingToVmInterface(macAddress, getVirtualMachineInterfaces(vm))
+			} else {
+				return err
+			}
+		}
+	}
+	if macMapChanged {
+		logger.V(1).Info("macMap is updated", "macPoolMap", p.macPoolMap)
+	}
+	return nil
+}
 
-		configMapUpdateNeeded := false
-		if configMap.Data == nil {
-			logger.V(1).Info("the configMap is empty", "configMapName", names.WAITING_VMS_CONFIGMAP, "macPoolMap", p.macPoolMap)
-			p.poolMutex.Unlock()
+func (p *PoolManager) getvmInstance(vmFullName string) (*kubevirt.VirtualMachine, error) {
+	vmFullNameSplit := strings.Split(vmFullName, "/")
+	vmNamespace := vmFullNameSplit[1]
+	vmName := vmFullNameSplit[2]
+	log.V(1).Info("getvmInstance", "vmNamespace", vmNamespace, "vmName", vmName)
+	if vmNamespace == "" || vmName == "" {
+		return nil, errors.New("failed to extract vm namespace and name")
+	}
+
+	requestUrl := fmt.Sprintf("apis/kubevirt.io/v1/namespaces/%s/virtualmachines/%s", vmNamespace, vmName)
+	log.V(1).Info("getvmInstance", "requestURI", requestUrl)
+
+	result := p.kubeClient.ExtensionsV1beta1().RESTClient().Get().RequestURI(requestUrl).Do(context.TODO())
+	if result.Error() != nil {
+		return nil, result.Error()
+	}
+	vm := &kubevirt.VirtualMachine{}
+	err := result.Into(vm)
+	if err != nil {
+		return nil, err
+	}
+
+	return vm, nil
+}
+
+func (p *PoolManager) handleStaleLegacyConfigMapEntries(parentLogger logr.Logger) error {
+	logger := parentLogger.WithName("handleStaleLegacyConfigMapEntries")
+	configMap, err := p.kubeClient.CoreV1().ConfigMaps(p.managerNamespace).Get(context.TODO(), names.WAITING_VMS_CONFIGMAP, metav1.GetOptions{})
+	if err != nil {
+		logger.Error(err, "failed to get config map", "configMapName", names.WAITING_VMS_CONFIGMAP)
+		return err
+	}
+
+	configMapUpdateNeeded := false
+	if configMap.Data == nil {
+		return nil
+	}
+
+	for macAddressDashes, allocationTime := range configMap.Data {
+		// legacy Configmap uses time.RFC3339.
+		t, err := time.Parse(time.RFC3339, allocationTime)
+		if err != nil {
+			// TODO: remove the mac from the wait map??
+			logger.Error(err, "failed to parse allocation time")
 			continue
 		}
 
-		for macAddress, allocationTime := range configMap.Data {
-			t, err := time.Parse(time.RFC3339, allocationTime)
+		logger.Info("data:", "configMapName", names.WAITING_VMS_CONFIGMAP, "configMap.Data", configMap.Data, "macPoolMap", p.macPoolMap)
+
+		if now().After(t.Add(time.Duration(p.waitTime) * time.Second)) {
```

Comment:
Some vars will make this more readeable.

Reply:
removed the function